### PR TITLE
Exposing CommandGUI:RunCommand and CommandConsole

### PR DESCRIPTION
**Pull request by arcburnergit**
This allows us to do make use of some of the features of the console within lua, such as loading events with ships after already having loaded a ship.

### CODE REVIEW

File: wiki/Lua-Hyperspace-Module.md [hunk 1]
```diff
@@ -1069,3 +1069,44 @@ local _, canMove = crew.extend:CalculateStat(Hyperspace.CrewStat.CAN_MOVE)
...
+
+### Methods
+- `bool :RunCommand(CommandGui *commandGui, const std::string& command)`
+   - Used for running a Hyperspace command.
+- `CommandConsole .GetInstance()`
+   - Returns the main instance of `CommandConsole`. Always use this to access any members belonging to this class.
+
+### Fields
+- `bool` `.enabled`
+   - Whether or not the Command Console is enabled or not, this is set in Hyperspace.xml.
```
Comment (ChronoVortex) [inline wiki/Lua-Hyperspace-Module.md:1084]: You have a redundant "or not" in this sentence, one of them needs to be removed.

File: wiki/Lua-Hyperspace-Module.md [hunk 2]
```diff
@@ -1069,3 +1069,44 @@ local _, canMove = crew.extend:CalculateStat(Hyperspace.CrewStat.CAN_MOVE)
...
+
+## CommandGUI
```
Comment (ChronoVortex) [inline wiki/Lua-Hyperspace-Module.md:1086]: The case of this struct's name is `CommandGui`, not `CommandGUI`.

File: wiki/Lua-Hyperspace-Module.md [hunk 3]
```diff
@@ -1069,3 +1069,44 @@ local _, canMove = crew.extend:CalculateStat(Hyperspace.CrewStat.CAN_MOVE)
...
+- `bool bAutoPaused`
+- `bool menu_pause`
+- `bool event_pause`
+- `Button upgradeButton`
+- `bool dangerLocation`
+- `bool bHideUI`
+- `bool jumpComplete`
+- `int mapId`
+- `bool secretSector`
+- `bool choiceBoxOpen`
```
Comment (ChronoVortex) [inline wiki/Lua-Hyperspace-Module.md:1112]: This doesn't follow the formatting of fields in the rest of the page. for example, `ShipStatus shipStatus` should be `ShipStatus` `.shipStatus`. If any of the classes of these fields are exposed and documented, there should also be links leading to their section in this page.

**Review by ChronoVortex — CHANGES_REQUESTED** [review]

**Comment by Nasa62** [comment]
@ChronoVortex I'm not sure how I feel about this.
I'd almost rather implement the console features in Lua and instead change the command console to a Lua input but with some backwards compatibility for the old commands rather than allowing Lua to hook into the console.

Kinda a catch-22, if we allow this, then Lua code can be forever tied to console commands, instead of exposing the actions those commands take to Lua and allowing us to clean up the console in the future so it's more of a proper scripting console with some hardcoded default commands like most other games would have first.

(so like the default input is lua, unless you use one of the special words that matches the old commands for compatibility)

I guess lua->console even when there's a "improved console" could still work but it feels weird... I feel like having those features accessible from Lua instead is better than running weird console commands via Lua.

**Pull request closed without merging**